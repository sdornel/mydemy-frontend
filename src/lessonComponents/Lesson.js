import React from 'react'
import {connect} from 'react-redux'
import {selectingLesson} from '../redux/actions'
import {Button, Grid} from 'semantic-ui-react'
import {Link} from 'react-router-dom'
import LessonVideo from './LessonVideo'

class Lesson extends React.Component {
    constructor(){
        super()
        this.state = {
            // courseId: "",
            lessonId: "",
        }
    }

    componentWillMount(){
        // debugger
        // let courseId = parseInt(this.props.match.params.courseId)
        // let lessonId = parseInt(this.props.match.params.lessonId)
        
        // console.log("lessonId", this.state.lessonId)

        // this.props.selectingLesson(this.props.lessonId)
        //     this.setState({
        //         // courseId: courseId,
        //         lessonId: this.props.lessonId
        // })
    }

    componentDidMount(){
        // debugger
        // this.props.selectingLesson(this.props.lessonId)
        let lessonId = this.props.lId
        this.setState({
            // courseId: courseId,
            // lessonId: this.props.lessonId
            lessonId: lessonId,
        })
    }

    render(){
        console.log("lesson state", this.state)
        console.log("lesson props", this.props)
        // debugger
        return !this.props.selectedLesson ? null : (
            <div>
                <br></br><br></br>
                <div className="lesson-content-div">
                    {this.props.selectedLesson.video !== null ? // use null/nil
                    <iframe width="560" height="315" src={this.props.selectedLesson.video} frameBorder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen>
                    </iframe>
                    : 
                    null}
                    <br></br><br></br>
                    {/* <p>{this.props.selectedLesson.text_content}</p> */}
                    <div className="lesson-content-text-area-div">
                        <textarea className="lesson-content-text-area" value={this.props.selectedLesson.text_content}></textarea>
                    </div>
                </div>
                <br></br><br></br><br></br><br></br><br></br><br></br><br></br><br></br><br></br><br></br><br></br><br></br>
            </div>
        // return !this.props.selectedLesson ? null : (
        // // return (
        //     <div>
        //     <br></br><br></br>
        //     <div className="lesson-content-div">
        //         {/* <iframe width="560" height="315" src={this.props.selectedLesson.video} title="video" frameBorder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen>
        //         </iframe> */}
        //         <LessonVideo lesson={this.props.selectedLesson}/>
        //         <br></br><br></br>
        //         <p>{this.props.selectedLesson.text_content}</p>
        //     </div>
        // </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        selectedLesson: state.selectedLesson,
        // lessons: state.lessons
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        selectingLesson: (info) => {dispatch( selectingLesson(info) )}
    }
}

// // export default Lesson

export default (connect(mapStateToProps, mapDispatchToProps)(Lesson));