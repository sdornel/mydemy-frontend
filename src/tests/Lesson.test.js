import React from 'react';
import {shallow} from 'enzyme';
import Lesson from '../lessonComponents/Lesson';

describe('Course', () => {
    const mockLesson = {
        id: 8,
        lesson_name: "Lesson One",
        text_content: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc auctor, ante sed euismod hendrerit, ex lacus maximus arcu, eu condimentum nibh ex a quam. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nullam gravida risus ex, ac aliquam elit gravida ac. Aliquam sodales gravida magna non dictum. Quisque maximus neque eget erat sollicitudin semper. Ut eu purus finibus leo dictum pellentesque a et dui. Aliquam egestas, sem at dignissim pulvinar, metus tellus gravida velit, ut iaculis metus tellus eget ex. Maecenas faucibus rhoncus feugiat. Sed vel diam at nunc luctus feugiat eu in justo. Integer vel dui ipsum. Pellentesque sodales nisi eu sem finibus, sed tristique libero auctor.",
        lesson_number: 8,
        video: "video",
    }
    it('should receive props', () => {
        const wrapper = shallow(<Lesson.WrappedComponent {...mockLesson} />)
        expect(wrapper).toMatchSnapshot()
    })
})