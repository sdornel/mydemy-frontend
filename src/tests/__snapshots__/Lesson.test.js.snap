// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course should receive props 1`] = `ShallowWrapper {}`;
